### DISCUSSION TITLE
PxWeb 2.0 - Do you have any ideas for the new user interface?

### DISCUSSION THREAD

**Discussion by AsaArrhen** [statisticssweden/PxWeb#314] [Ideas]
We intend to start sketching the new user interface for PxWeb 2.0 in January 2023. Have you done any user survey or got any user insights? Please let us know! Put your idea here before January.

Åsa Arrhén, SCB & Kristin Glomsås, SSB

**Comment by larpSTATGL** [comment]
In your search for new interfaces, please don't forget the present as important baseline. Users who have taken the time to master the present, will oppose to changes. 

Good luck 

:-)

**Comment by emilwest** [comment]
One thing I would like to see specifically in the tableview interface is an easier way to visualize cellnotes. 
I personally like the Eurostat data browser because it prints Eurostat flags directly in the cells, with explanations in the footer:
 
![image](https://user-images.githubusercontent.com/11556959/198519944-95d30b61-4a7a-41a0-8724-4ef22b1e5a92.png)

For example, in px we often use:

* ‘..’ for missing values 

 * ‘.’ for ‘not applicable’ 

* and sometimes ‘-’ is used to mean exactly zero. 


At Nordic Statistics we conducted a survey on this, and we found out that the NSI:s use and interpret these differently. Therefore it would be good to have the possibility in pxweb to define at least what the special characters mean (‘.’,’..’,’…’, ‘-’, etc) and show it directly below the table in a footer (not footnotes). Currently Statistics Faroe Islands show this in their footnotes, but that requires that you add this to all footnotes for all px-files, it would be great if this could be defined only once in pxweb instead. 
Another request is to make cellnotes hoverable, so that you can display the text when you hover the mouse over a value, like for example ‘break in time series’. Currently you must click on a cell to show the text.

**Comment by Liga-Saleniece** [comment]
There are cases when it would be very useful to see variable names in the data table above values. Maybe it could be an additional option/button to show variable names? The example below shows a data table that uses the same territorial units under two different variables. Because the variable names are'nt visible, we indicated next to the value its belonging to variable. With red text, I visualized what it might look like if the variable names were visible.

![pxweb_variables](https://user-images.githubusercontent.com/116885196/198637841-b531470f-3180-4c0a-9cca-1912c02c0bf3.png)

**Comment by Liga-Saleniece** [comment]
It's very nice that column headings freeze for large tables in fullscreen view. It would be nice if row headings would freeze as well. But if it also works on normal view, it would be excellent.

When selecting indicators, there have been cases when data users do not notice the grouping menu under the variable. We would appreciate it if it were more highlighted and visible.

**Comment by salgo60** [comment]
* [#2](https://github.com/salgo60/statisticssweden/issues/2) Linked data

I would like to see [Linked data](https://en.wikipedia.org/wiki/Linked_data) (see [5 star data](https://5stardata.info/en/)) and [FAIRDATA](https://www.go-fair.org/fair-principles) very important is [FAIRDATA F1](https://www.go-fair.org/fair-principles/f1-meta-data-assigned-globally-unique-persistent-identifiers/) to get [persistant unique identifiers](https://en.wikipedia.org/wiki/Persistent_identifier)

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/14206509/204111221-314baa91-fac0-4ca2-b8a1-e75b81d12603.png">

**Example:** sv:Wikipedia helped SCB to publish "Kommun info" in 2022 see [T266863](https://phabricator.wikimedia.org/T266863) as SCB had "Kommun kod" as a "key" to the data it was done rather straight on **BUT  "Kommun kod" is not persistant**

1) the problem with "Kommun kod" its not persistant as its based on the county a Swedish Municipal belong to see [sv:Wikipedia](https://sv.wikipedia.org/wiki/Kommunkod)

> Vid länssammanslagningar, som i fallen [Skåne län](https://sv.wikipedia.org/wiki/Sk%C3%A5ne_l%C3%A4n) och [Västra Götalands län](https://sv.wikipedia.org/wiki/V%C3%A4stra_G%C3%B6talands_l%C3%A4n) , har vissa kommuner fått nya koder.

In Wikipedia we have a layer [Wikidata](https://www.youtube.com/watch?v=m_9_23jXPoE) that contains structured data --> 

* [List Swedish municipalities that has changed english](https://w.wiki/44V6) [swedish](https://w.wiki/43ot) changed "kommunkod" 
  * this question is also important for the [project Nationell Dataverkstad #21](https://github.com/salgo60/Anslagstavla/issues/21) - we dont see them deliver 5 star data today [#6](https://github.com/salgo60/Anslagstavla/issues/6)

![image](https://user-images.githubusercontent.com/14206509/204084662-b97f7d75-3831-4513-88f2-d1695c30e615.png)

* Wikidata [Bengtfors kommun Q267462](https://www.wikidata.org/wiki/Q267462#P525)
<img width="1010" alt="image" src="https://user-images.githubusercontent.com/14206509/204111314-d802e0aa-8553-4018-8285-2f3c8f2c6c4c.png">

Dont hesitate to answer here or contact me see [my CV](https://github.com/salgo60/StorsthlmWIkidata/issues/1#issuecomment-1252326546) or backlogs we create to track problems like above we find
* ["Riksdagen"](https://github.com/salgo60/Wikidata_riksdagen-corpus/issues/50)
* [DIGG](https://github.com/salgo60/DiggUptime/issues/47)
* ["Nationell dataverkstad"](https://github.com/salgo60/Anslagstavla/issues/3)

**Comment by KNaumanis** [comment]
Some insights and ideas from Statistics Estonia.

1.	Hierarchies – sometimes in a dataset it’s necessary to define more than one dimension as hierarchical (including Indicator). That feature increases user friendliness. It’s important for the database user to see the relationships between the elements in the code list and different levels in the code list.
•	Variable selection: when developing the hierarchy functionality, it’s important to keep in mind providing the option for user to manipulate the selection of code list elements of different levels in the code list – select all, unselect all, select level, unselect level, etc.
•	Table view: it’s also necessary to view the hierarchy (relations between the code list elements) in the table view. With the possibility to hide and show the levels.

2.	Drop down lists – when a dimension is presented as a drop-down list it would be great if user could save the selection of dimension elements in different groups and then to display all the selected elements (from all the different groups) in the (final) table. 

3.	A thorough and well explained documentation.
